src/main/java/com/hazelcast/kubernetes/HazelcastKubernetesDiscoveryStrategy.java

```diff
@@ -99,6 +116,36 @@ public void start() {
         endpointResolver.start();
     }
 
+    @Override
+    public Map<String, Object> discoverLocalMetadata() {
+        if (memberMetadata.isEmpty()) {
+            memberMetadata.put(PartitionGroupMetaData.PARTITION_GROUP_ZONE, discoverZone());
+        }
+        return memberMetadata;
+    }
+
+    private String discoverZone() {
+        try {
+            String podName = System.getenv("POD_NAME");
+            if (podName == null) {
+                podName = System.getenv("HOSTNAME");
+            }
+            if (podName == null) {
+                podName = InetAddress.getLocalHost().getHostName();
+            }
+            String zone = client.zone(namespace, podName);
+            if (zone != null) {
```

Comment:
is it possible that k8s api returns null?

Reply:
Yes, is the label is not defined (e.g. you run Kubernetes locally which doesn't set it), then `client.zone()` returns `null`.

Reply:
ok